extensions/amp-mega-menu/amp-mega-menu.md

```diff
@@ -0,0 +1,228 @@
+
+[example playground="true" imports="amp-mega-menu,amp-list" template="amp-mustache"]
+```html
+<amp-mega-menu height="60" layout="fixed-height">
+  <amp-list height="350" layout="fixed-height" src="{{server_for_email}}/static/samples/json/product-single-item.json" single-item>
+    <template type="amp-mustache">
+      <nav>
+        <ul>
+          {{#values}}
+          <li>
+            <h4 role="button">{{name}}</h4>
+            <div role="dialog">
+              <amp-img src="{{server_for_email}}{{img}}" width="320" height="213"></amp-img>
+              <p>Price: $<b>{{price}}</b></p>
+            </div>
+          </li>
+          {{/values}}
+        </ul>
+      </nav>
+    </template>
+  </amp-list>
+</amp-mega-menu>
+```
+[/example]
+
+Here is the JSON file that we used:
+
+```json
+{
+  "items": [{
+    "values": [{
+      "id": 1,
+      "img": "/static/samples/img/product1_640x426.jpg",
+      "name": "Apple",
+      "price": "1.99"
+    }, {
+      "id": 2,
+      "img": "/static/samples/img/product2_640x426.jpg",
+      "name": "Orange",
+      "price": "0.99"
+    }, {
+      "id": 3,
+      "img": "/static/samples/img/product3_640x426.jpg",
+      "name": "Pear",
+      "price": "1.50"
+    }]
+  }]
+}
+```
+
+Note that the `<nav>` element must be parented by either the `<amp-mega-menu>` component itself or a `<template>`.
+
+### Responsive design with `<amp-sidebar>`
+
+For viewports that may be too narrow to display the menu items in one row, we recommend placing the menu inside an [amp-sidebar](https://amp.dev/documentation/components/amp-sidebar/) instead and responsively switch between the two components via media query.
+
+The following sample demonstrates how to hide the mega menu when viewport width is less than `960px` and instead show a button to open the sidebar.
+
+```html
+<head>
+  ...
+  <style amp-custom>
+    @media (max-width: 959px) {
+      #mega-menu {
+        display: none;
+      }
+    }
+    @media (min-width: 960px) {
+      .sidebar-open-btn {
+        display: none;
+      }
+    }
+  </style>
+  ...
+</head>
+```
+```html
+<body>
+  <header>
+    <amp-mega-menu id="mega-menu" height="50" layout="fixed-height">
+      <nav>
+      ... <!-- list of menu items here -->
+      </nav>
+    </amp-mega-menu>
+    <button class="sidebar-open-btn" on="tap:sidebar"></button>
+  </header>
+  <amp-sidebar id="sidebar" layout="nodisplay">
+    ... <!-- list of menu items here -->
+  </amp-sidebar>
+  ...
+</body>
+```
+
+## Attributes
+
+<table>
+  <tr>
+    <td width="40%"><strong>data-close-button-aria-label (optional)</strong></td>
+    <td>Optional attribute used to set ARIA label for the close buttons added for accessibility.</td>
+  </tr>
+  <tr>
+    <td width="40%"><strong>common attributes</strong></td>
+    <td>This element includes <a href="https://amp.dev/documentation/guides-and-tutorials/learn/common_attributes">common attributes</a> extended to AMP components.</td>
+  </tr>
+</table>
+
```

Comment:
Add a section on Accessibility. In particular, reference the accessibility documentation with regards to Hover, as well as explain the elements and attributes added for accessibility.